src/components/modal/modal.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import {SBComponent} from 'stylable-react-component';
+import {root} from 'wix-react-tools';
+import {Portal} from '../../../src';
+import {noop} from '../../utils';
+import {enableScrolling, stopScrolling} from '../../utils/stop-scrolling';
+import styles from './modal.st.css';
+
+export interface RequestCloseEvent {
+    source: string;
+}
+
+export interface ModalProps extends React.HTMLAttributes<HTMLDivElement> {
+    isOpen: boolean;
+    onRequestClose?({source}: RequestCloseEvent): void;
+}
+
+@SBComponent(styles)
+export class Modal extends React.PureComponent<ModalProps, {}> {
+    public static defaultProps: ModalProps = {
+        isOpen: false,
+        onRequestClose: noop
+    };
+
+    public componentDidMount() {
+        this.shouldEnableScrolling(!this.props.isOpen);
+    }
+
+    public componentWillReceiveProps(nextProps: ModalProps) {
+        this.shouldEnableScrolling(!nextProps.isOpen);
```

Comment:
Why not call this on componentDidUpdate? Also I don't this code is a tested, I commented it out and all the tests pass, but the demo page allows you to scroll when the modal is open.